tools/txs/src/txs_cli_community.rs

```diff
@@ -82,6 +95,189 @@ impl ProposeTx {
     }
 }
 
+#[derive(clap::Args)]
+pub struct BatchTx {
+    #[clap(short, long)]
+    /// The Community Wallet to schedule transaction
+    pub community_wallet: AccountAddress,
+    #[clap(short, long)]
+    /// JSON file with batch payments
+    pub file: PathBuf,
+    #[clap(short, long)]
+    /// Write the result json to a different file (otherwise will overwrite)
+    pub out: Option<PathBuf>,
+    #[clap(long)]
+    /// Just check if the destinations are slow wallets
+    pub check: bool,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+struct ProposePay {
+    recipient: String,
+    parsed: Option<AccountAddress>,
+    amount: u64,
+    description: String,
+    is_slow: Option<bool>,
+    proposed: Option<bool>,
+    voters: Option<Vec<AccountAddress>>,
+    error: Option<String>,
+    note: Option<String>,
+}
+
+// DEV NOTE: really what we should be doing is creating a Move transaction
+// script that submits all TXS in a batch and executes all or aborts
+// (an atomic batch).
+impl BatchTx {
+    pub async fn run(&self, sender: &mut Sender) -> anyhow::Result<()> {
+        let data = fs::read_to_string(&self.file).expect("Unable to read file");
+        let mut list: Vec<ProposePay> = serde_json::from_str(&data).expect("Unable to parse");
+
+        let ballots =
+            account_queries::multi_auth_ballots(sender.client(), self.community_wallet).await?;
+        let d = ballots.as_object().unwrap();
+        let v = d.get("vote").unwrap().as_object().unwrap();
+        let p = v.get("ballots_pending").unwrap().as_array().unwrap();
+
+        let mut pending: HashMap<AccountAddress, ProposePay> = HashMap::new();
+        p.iter().for_each(|e| {
+            let o = e.as_object().unwrap();
+            let prop = o.get("tally_type").unwrap().as_object().unwrap();
+            let data = prop.get("proposal_data").unwrap().as_object().unwrap();
+            let recipient: AccountAddress = data
+                .get("payee")
+                .unwrap()
+                .as_str()
+                .unwrap()
+                .parse()
+                .unwrap();
```

Comment:
Poster child for why FP is evil. Would it be possible to wrap this chain in a helper fn?